tests/test.py

```diff
@@ -450,7 +463,16 @@ def test_variant(self):
         self.assertEqual(1, len(n.get_variant_ids()))
 
 
-
+    def test_monitor_state(self):
+        n = pp.network.create_eurostag_tutorial_example1_network()
+        sa = pp.security.create_analysis()
+        sa.add_single_element_contingency('NHV1_NHV2_1', 'NHV1_NHV2_1')
```

Comment:
I have same concern with sensitivity factors per contingency that is not yet interfaced in Python.
Instead of having the enum on Python side I would have only rely on a contingencies array with the following convention:
  - None -> all
  - empty -> none
  - non empty array -> specific for theses contingencies

@EtienneLt @sylvlecl what do you think?

Reply:
Indeed it looks more convenient and efficient to be able to specify a list of contingencies instead of just one.

But for the convention `None` = all and `[]` = N, I don't think it will be clear for the user !
To improve this we propose to:
* keep the enum
* add convenience methods for the user which will call the enum-based methods, see for example below. I would propose to rename to "add monitored elements" which sounds more intuitive than "add state monitors", what do you think about it ?
```python
sa.add_precontingency_monitored_elements(...)
sa.add_postcontingency_monitored_elements(contingency_ids=[...], ...)
sa.add_all_states_monitored_elements(...)
```